app/features/signup/verify-email-form.tsx

```diff
@@ -82,10 +93,32 @@ export function VerifyEmailForm() {
     }
   };
 
-  if (user.email_verified || isGuestUser) {
+  useEffect(() => {
+    if (code) {
+      const handleVerificationCode = async () => {
```

Comment:
So this will only auto verify the code from the params?

 Should we also auto-check/verify when the form value changes? Most users that don't click the link will probably be copy pasting. If we know the length/format of a valid code then we can wait to send the `verifyEmail` request until the code is of the correct format; although, this could be messy if the code format changes.

Reply:
sure we can implement that as well